src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -138,8 +138,8 @@ def extract_strings(self) -> Tuple[Mapping[int, Set[str]], Set[str]]:
     @metrics.wraps("process_messages.reconstruct_messages")
     def reconstruct_messages(
         self,
-        mapping: Mapping[int, Mapping[str, int]],
-        bulk_record_meta: Mapping[str, Tuple[int, FetchType]],
+        mapping: Mapping[int, Mapping[str, Optional[int]]],
```

Comment:
Please let's have a unit test for this method. 
There are at least 5 code paths to cover:
- all good, all strings resolved
- metric id rate limited 
- some tag keys rate limited
- some tag values rate limited
- the same string is successful in an organization and not in another (two different messages). 

While the end to end test for the `process` method is good and useful, you would not test each and every one of those code paths above in an end to end test. This is one of the advantage of breaking this logic out of the massive process method we had before. Now we can test each case separately without going crazy.